webui/react/src/components/kit/Icon.tsx

```diff
@@ -117,11 +254,28 @@ const Icon: React.FC<Props> = ({
 }: Props) => {
   const classes = [css.base];
 
-  if (name) classes.push(`icon-${name}`);
+  const rowIcon = useMemo(() => {
+    if (name.includes('small')) return <RowIconSmall size={size} title={title} />;
+    if (name.includes('large')) return <RowIconLarge size={size} title={title} />;
+    if (name.includes('xl')) return <RowIconExtraLarge size={size} title={title} />;
+
+    return <RowIconMedium size={size} title={title} />; // returns medium by default
+  }, [name, size, title]);
+  const svgIcon = useMemo(() => {
+    if (name === 'columns') return <ColumnsIcon size={size} title={title} />;
+    if (name === 'filter') return <FilterIcon size={size} title={title} />;
+    if (name === 'options') return <OptionsIcon size={size} title={title} />;
+    if (name === 'panel') return <PanelIcon size={size} title={title} />;
+    if (name.includes('row')) return rowIcon;
+
+    return null;
+  }, [rowIcon, name, size, title]);
+
+  if (name && name !== 'filter') classes.push(`icon-${name}`);
```

Comment:
this check is unnecessary, no?

Reply:
the `classes` array is used in the `span` for the fonticons, but if `name` is `filter`, `svgIcon` is nonnull, so it's unused.

Seems like this part of the code is difficult to reason about. Consider splitting the component into two: one that renders the icon, and the other that wraps the icon in the tooltip. That way, you can bail early from the render when rendering svgs. In addition, is overriding the preexisting `filter` icon intentional?

Reply:
Why did you add this check back?

Reply:
Ok, so `icon-filter` actually adds an icon via css? Can you add a comment explaining that?

Reply:
This line is still present though